### ISSUE TITLE
options.defaultLayout not used?

### ISSUE THREAD

**Issue by macklinu** [ChristopherBiscardi/gatsby-mdx#34]
I'm working on a design system documentation site and am trying to use gatsby-mdx and `options.defaultLayout` to use the same layout file across all `.mdx` pages in my project. However, the default layout doesn't appear to be rendering.

Here are the steps I took and a [repo that reproduces the issue](https://github.com/macklinu/gatsby-mdx-default-layout-issue).

- Create a new gatsby v2 site: `npx -p gatsby-cli gatsby new site https://github.com/gatsbyjs/gatsby-starter-default\#v2`
- Install gatsby-mdx from the newly created gatsby site directory: `yarn add gatsby-mdx @mdx-js/mdx @mdx-js/tag`
- Add the gatsby-mdx plugin to the gatsby config:

```js
module.exports = {
  plugins: [
    {
      resolve: 'gatsby-mdx',
      options: {
        defaultLayout: require.resolve('./src/components/layout.js'),
      },
    },
  ],
}
```

- Replace `src/pages/index.js` with `src/pages/index.mdx` and add any MDX content
- Run `yarn develop` to run the gatsby dev server

**Expected**: Visiting http://localhost:8000 wraps the `src/pages/index.mdx` content in `options.defaultLayout` (`src/components/layout.js`)
**Actual**: Visiting http://localhost:8000 results in the MDX content rendered without any wrapping element

Is my simplified pseudocode understanding of `defaultLayout` correct?

```js
function gatsbyMdxPlugin(source, options) {
  let code = mdx(source)
  if (options.defaultLayout) {
    let Layout = require(options.defaultLayout)
    render(<Layout>{code}</Layout>)
  } else {
    render(code)
  }
}
```

**Comment by ChristopherBiscardi** [comment]
Thanks for the report and the lovely repro steps! I've fixed and published a fix in 0.0.5. You should be able to upgrade and see the fix :) The bug was caused by a reordering of some position-sensitive code during a refactor.

---

The logic for default layouts is roughly:

```js
const code = mdx(file);

if(hasDefaultExport(code)) {
  // do nothing; a default export means there's already a layout
} else if (options.defaultLayout) {
  // set default export in code to defaultLayout
}
```

**Comment by macklinu** [comment]
@ChristopherBiscardi thanks for your help - you rock! Confirmed the issue is no longer present in v0.0.5. 😄
